### ISSUE TITLE
Boxplot won't sort on op result

### ISSUE THREAD

**Issue by metasoarous** [vega/vega-lite#6848]
I'm attempting to sort a set of boxplots by either the median or mean of the quantitative field, but it's not working. The docs suggest this should work: https://vega.github.io/vega-lite-v3/docs/sort.html#sort-by-a-different-field

Spec looks something like this:

```
{"data": {"values": ...},
  "mark": "boxplot",
  "encoding": {
    "x": {
      "field": "category",
      "sort": {"op": "median", "field": "blah"}
    },
    "y": {"field": "blah", "type": "quantitative"}
}
```

Filing a bug report, since there doesn't seem to be anything in the docs above that suggest this shouldn't work for boxplots. If this is expected though, can we please consider this issue a feature request?

Thanks as always!

**Comment by metasoarous** [comment]
Checking the console, I see

```
property {"field":"blah","op":"median"} as unioned domains only support boolean or op "count", "min", and "max".
```

So I'm assuming this is indeed not supported.

Would you please help me understand why this isn't possible for unioned domains? Is this a fundamental limitation, or one for which a feature request might be considered?

Thanks again!

**Comment by domoritz** [comment]
It's a somewhat fundamental limitation (not specific to Vega or Vega-Lite) but there are many special cases (like this one) where sorting can work. See https://github.com/vega/vega-lite/issues/5048.

**Comment by willcrichton** [comment]
@metasoarous were you able to figure out a fix to this? I read through the links in #5048 that @domoritz but couldn't arrive at a solution short of manually computing the median and giving a sort that way.

**Comment by metasoarous** [comment]
@willcrichton Yes, I did get this to work by passing a list of values to the `sort` property of the encoding. I did something along the lines of

```
{"data": {"values": ...},
  "mark": "boxplot",
  "encoding": {
    "x": {
      "field": "category",
      "sort": ["category1", "category2", etc]
    },
    "y": {"field": "blah", "type": "quantitative"}
}
```
